data/lib/core/player.lua

```diff
@@ -698,3 +698,140 @@ end
 function Player.getAccountStorageValue(self, key)
 	return Game.getAccountStorageValue(self:getAccountId(), key)
 end
+
+function Player.getMaxVIPEntries(self)
+	local groupMaxVipEntries = self:getGroup():getMaxVipEntries()
+	if groupMaxVipEntries > 0 then
+		return groupMaxVipEntries
+	end
+	return configManager.getBoolean(self:isPremium() and configKeys.VIP_PREMIUM_LIMIT or configKeys.VIP_FREE_LIMIT)
+end
+
+function Player.addVIP(self, name)
+	if string.len(name) > PLAYER_NAME_LENGTH then
+		return
+	end
+
+	local vipGuid
+	local vipName
+	local status
+
+	-- check if the VIP is online
+	local vipPlayer = Player(name)
+	if vipPlayer then
+		if vipPlayer:hasFlag(PlayerFlag_SpecialVIP) and not self:hasFlag(PlayerFlag_SpecialVIP) then
+			self:sendTextMessage(MESSAGE_STATUS_SMALL, "You can not add this player.")
+			return
+		end
+
+		vipGuid = vipPlayer:getGuid()
+		vipName =  vipPlayer:getName()
+		status = (not vipPlayer:isInGhostMode() or self:canSeeGhostMode(vipPlayer)) and VIPSTATUS_ONLINE or VIPSTATUS_OFFLINE
+	else
+		-- if not online, attempt to load by name
+		local resultId = db.storeQuery("SELECT `name`, `id`, `group_id` FROM `players` WHERE `name` = " .. db.escapeString(name))
+		if not resultId then
+			self:sendTextMessage(MESSAGE_STATUS_SMALL, "A player with this name does not exist.")
+			return
+		end
+		
+		-- find vip group and check if exists
+		local groupId = result.getNumber(resultId, "group_id")
+		local group = Group(groupId)
+		if not group then
+			return
+		end
+
+		if group:hasFlag(PlayerFlag_SpecialVIP) and not self:hasFlag(PlayerFlag_SpecialVIP) then
+			self:sendTextMessage(MESSAGE_STATUS_SMALL, "You can not add this player.")
+			return
+		end
+
+		vipGuid = result.getNumber(resultId, "id")
+		vipName = result.getString(resultId, "name")
+		status = VIPSTATUS_OFFLINE
+	end
+
+	local playerVIP = VIP(self:getGuid())
+	if not playerVIP:canAdd() then
+		self:sendTextMessage(MESSAGE_STATUS_SMALL, "You cannot add more buddies.")
+		return
+	end
+
+	if playerVIP:has(vipGuid) then
+		self:sendTextMessage(MESSAGE_STATUS_SMALL, "This player is already in your list.")
+		return
+	end
+
+	local description = ""
+	local icon = 0
+	local notify = false
+
+	local accountId = self:getAccountId()
+	db.asyncQuery("INSERT INTO `account_viplist` (`account_id`, `player_id`, `description`, `icon`, `notify`) VALUES (" .. accountId .. ", " .. vipGuid .. ", " .. db.escapeString(description) .. ", " .. icon .. ", " .. (notify and 1 or 0) .. ")")
```

Comment:
These are the defaults

```suggestion
	local accountId = self:getAccountId()
	db.asyncQuery("INSERT INTO `account_viplist` (`account_id`, `player_id`) VALUES (" .. accountId .. ", " .. vipGuid .. ")")
```